overlord/auth/auth.go

```diff
@@ -81,3 +82,30 @@ func User(st *state.State, id int) (*UserState, error) {
 	}
 	return nil, fmt.Errorf("invalid user")
 }
+
+// Authenticator returns MacaroonAuthenticator for current authenticated user represented by UserState
+func (us *UserState) Authenticator() *MacaroonAuthenticator {
+	return newMacaroonAuthenticator(us.Macaroon, us.Discharges)
+}
+
+// MacaroonAuthenticator is a store authenticator based on macaroons
+type MacaroonAuthenticator struct {
+	Macaroon   string
+	Discharges []string
+}
+
+func newMacaroonAuthenticator(macaroon string, discharges []string) *MacaroonAuthenticator {
+	return &MacaroonAuthenticator{
+		Macaroon:   macaroon,
+		Discharges: discharges,
+	}
+}
+
+// Authenticate will add the store expected Authorization header for macaroons
+func (ma *MacaroonAuthenticator) Authenticate(r *http.Request) {
```

Comment:
`var buf bytes.Buffer`
then below use
`fmt.Fprintf(&buf, ...)`
and finally
`r.Header.Set("...", buf.String())`